### ISSUE TITLE
Humpback - AIs can fail to navigate stairs

### ISSUE THREAD

**Issue by lewri** [Regalis11/Barotrauma#10206]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Sometimes I've noticed in playing Humpback the AI crew is not able to get down the stairs to the bottom floor. They appear to try to get down but instead run left and right of the top of the stairs. This seems to happen mostly when they are on cleanup duties, or I've asked one to follow me.

Sometimes I can fix them by grabbing them, pulling them away from the stairs, then letting go. Though not always a success.

If I remember correctly, back when I played earlier in the year (or 2021) this issue wasn't present.

### Reproduction steps

1. Load the game with Humpback, crew, and items to put away.
2. Assign some crew to cleanup duties.
3. Observe stairwell to see if any get stuck.

Alternatively:
1. Ask a crew member to follow you
2. Go down the stairs and see if they get stuck.

### Bug prevalence

Happens regularly

### Version

0.19.14.0

### Which operating system did you encounter this bug on?

Windows

**Edit 20/11 -- Not sure if relevant, but my campaign has been through a few years of version changes, though nothing has ever seemed broken until now**

**Comment by lewri** [comment]
I finally got round to recording this -- sorry

[screen-recording(5).webm](https://user-images.githubusercontent.com/20030128/200915124-3f0ec623-c416-4bf1-855a-8f595b433801.webm)

**Comment by itchyOwl** [comment]
Can't repro this. Have you tested if you can reproduce the issue with a fresh campaign?

**Comment by lewri** [comment]
> Can't repro this. Have you tested if you can reproduce the issue with a fresh campaign? There's been some adjustments to the waypoints and to the stair navigation logic what I suspect could cause it on a sub with waypoints generated with the old logic.

May be likely. Would that mean the waypoints didn't convert correctly when the game updated?

**Comment by itchyOwl** [comment]
Yeah, it's like I suspected. Old submarines have old waypoints and there's not really a fix for an old saved game, other than switching the sub.

This is your sub. The bots can get stuck to the waypoints there (for a while):
![image](https://user-images.githubusercontent.com/5370738/214326062-f2258617-3c50-4d39-9dfc-c6640cf21d19.png)

When I adjusted the navigation logic a while ago, I also changed how the waypoints are generated. The generator should now create the waypoints on stairs like this:
![image](https://user-images.githubusercontent.com/5370738/214327440-95113469-fa41-4f59-86be-9acd7f6d097b.png)


Closing the ticket, because the issue is not fixable.
